.github/actions/setup/action.yml

```diff
@@ -0,0 +1,22 @@
+name: Setup
+runs:
+  using: composite
+  steps:
+    - uses: pnpm/action-setup@v2
+      with:
+        version: 7.18.2
+
+    - uses: actions/setup-node@v3
+      with:
+        node-version-file: '.nvmrc'
+        registry-url: https://registry.npmjs.org
+        cache: pnpm
+
+    - name: Install Foundry
+      uses: foundry-rs/foundry-toolchain@v1
+      with:
+        version: nightly
+
+    - name: Install node modules
+      run: pnpm install --frozen-lockfile
+      shell: bash
```

Comment:
We could further cache this by reading from cache if package.jsons and pnpm-lock haven't changed.  I can add that in it's own pr

Reply:
is `shell: bash` needed here? what happens without it?